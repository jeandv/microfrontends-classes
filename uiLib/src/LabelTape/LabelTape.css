.ui-label {
  width: 75px;
  padding: 4px;
  font-size: 10px;
  text-align: center;
  transform: rotate(-45deg) translate(-24px, -6px);
  position: absolute;
  top: 0;
}

.ui-label--4dx {
  background-color: #9bf;
  color: #fff;
}

.ui-label--promotions {
  background-color: #fc0;
}

.ui-label--discount {
  background-color: #0ac170;
  color: #fff;
  font-weight: 600;
}