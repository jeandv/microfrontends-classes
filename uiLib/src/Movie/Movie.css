.ui-movie {
  width: 24px;
  font-family: monospace;
}

.ui-movie-img {
  height: 230px;
  width: 100%;  
  }

.ui-movie-img>img {
  width: 100%;
}

.ui-movie-title {
  color: #000;
  font-weight: 700;
  text-align: center;
}