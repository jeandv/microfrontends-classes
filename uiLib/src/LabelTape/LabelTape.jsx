import React from 'react';
import PropTypes from 'prop-types';
import './LabelTape.css';

function LabelTape({ label, variant, ...props }) {
  const mode = variant ? ` ui-label--${variant}` : null;

  return (
    <div className={'ui-label' + mode} {...props}>
      {label}
    </div>
  );
}

LabelTape.propTypes = {
  /**
   * What color should the button be?
   */
  variant: PropTypes.oneOf(['discount', 'promotions', '4dx']),
  /**
   * Label text (Required)
   */
  label: PropTypes.string.isRequired,
};

LabelTape.defaultProps = {
  variant: 'discount',
  label: 'Descuentos',
};

export default LabelTape;
