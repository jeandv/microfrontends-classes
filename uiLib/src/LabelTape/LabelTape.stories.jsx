import React from 'react';

import LabelTape from './LabelTape';

export default {
  title: 'ui/LabelTape',
  component: LabelTape,
};

const Template = (args) => <LabelTape {...args} />;

export const Primary = Template.bind({});

// Adding Args
// Primary.args = {
//   primary: true,
//   label: 'Comprar',
// };

// export const Secondary = Template.bind({});

// Secondary.args = {
//   primary: true,
//   label: 'Comprar',
// };
