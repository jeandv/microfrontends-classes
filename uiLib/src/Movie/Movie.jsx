import React from 'react';
import PropTypes from 'prop-types';

import LabelTape from '../LabelTape/LabelTape';

import './Movie.css';

export default function Movie({
  variant,
  label,
  movieTitle,
  imgUrl,
  ...props
}) {
  return (
    <div className='ui-movie' {...props}>
      <div className='ui-movie-img'>
        <img src={imgUrl} alt={movieTitle} className='ui-movie-img-asset' />
        {variant != undefined && <LabelTape label={label} variant={variant} />}
      </div>
      <div
        className={['ui-movie-title', `ui-movie-title--${variant}`].join(' ')}
      >
        <h4 className='ui-movie-title-text'>{movieTitle}</h4>
      </div>
    </div>
  );
}

Movie.propTypes = {
  /**
   * Defines label background colors.
   */
  variant: PropTypes.oneOf(['discount', 'promotions', '4dx']),
  /**
   * Label text
   */
  label: PropTypes.string,
  /**
   * Movie title (Required)
   */
  movieTitle: PropTypes.string.isRequired,
};

// Movie.defaultProps = {
//   variant: 'promotions',
//   imgUrl:
//     'https://images.complex.com/complex/images/c_fill,dpr_auto,f_auto,q_auto,w_1400/fl_lossy,pg_1/gdv2pu6io6ekpg5r8mta/back-to-the-future?fimg-ssr-default',
//   movieTitle: 'Movie title',
//   label: 'Promociones',
// };
