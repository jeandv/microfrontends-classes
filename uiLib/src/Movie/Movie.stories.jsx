import React from 'react';

import Movie from './Movie';

export default {
  title: 'ui/Movie',
  component: Movie,
};

const CartelTemplate = (args) => <Movie {...args} />;

export const Primary = CartelTemplate.bind({});

// Cartel.args = {
//   label: 'Promociones',
//   variant: undefined,
//   movieTitle: 'Movie 01',
//   src: 'https://static.cinepolis.com/img/peliculas/41818/1/1/41818.jpg',
// };
