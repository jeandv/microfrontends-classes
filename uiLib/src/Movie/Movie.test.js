import React from 'react';
import renderer from 'react-test-renderer';
import Movie from './Movie';

const poster = 'https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/56c72e90-7908-4970-8e1b-b5b7098f4fda/df4gi93-bb27d107-8b1f-4dec-97f1-815424419af6.png/v1/fill/w_1280,h_1811,q_80,strp/john_wick_4___poster_v1_by_pyramid3_df4gi93-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9MTgxMSIsInBhdGgiOiJcL2ZcLzU2YzcyZTkwLTc5MDgtNDk3MC04ZTFiLWI1YjcwOThmNGZkYVwvZGY0Z2k5My1iYjI3ZDEwNy04YjFmLTRkZWMtOTdmMS04MTU0MjQ0MTlhZjYucG5nIiwid2lkdGgiOiI8PTEyODAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.xsATywrHQvyQFggEG8sGHffq7CApdMWZn7FJ9C_CgEA'

//comprobamos que renderice
it('render with label', () => {

  // creamos desde el renderer el movie component y lo convertimos a un JSON
  let movieTree = renderer.create(
    <Movie
      movieTitle='John Wick 4'
      imgUrl={poster}
      variant='4dx'
      label='4DX' />
  ).toJSON();

  // se hace el snapshot de dicho componente para mostrar como se veria con la funcion que queramos probar
  expect(movieTree).toMatchSnapshot();

});

it('render without label', () => {

  let movieTree = renderer.create(<Movie movieTitle='John Wick 4' imgUrl={poster} />).toJSON();

  expect(movieTree).toMatchSnapshot();

});