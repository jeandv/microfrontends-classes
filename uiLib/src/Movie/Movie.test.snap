// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with label 1`] = `
<div
  className="ui-movie"
>
  <div
    className="ui-movie-img"
  >
    <img
      alt="John Wick 4"
      className="ui-movie-img-asset"
      src="https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/56c72e90-7908-4970-8e1b-b5b7098f4fda/df4gi93-bb27d107-8b1f-4dec-97f1-815424419af6.png/v1/fill/w_1280,h_1811,q_80,strp/john_wick_4___poster_v1_by_pyramid3_df4gi93-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9MTgxMSIsInBhdGgiOiJcL2ZcLzU2YzcyZTkwLTc5MDgtNDk3MC04ZTFiLWI1YjcwOThmNGZkYVwvZGY0Z2k5My1iYjI3ZDEwNy04YjFmLTRkZWMtOTdmMS04MTU0MjQ0MTlhZjYucG5nIiwid2lkdGgiOiI8PTEyODAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.xsATywrHQvyQFggEG8sGHffq7CApdMWZn7FJ9C_CgEA"
    />
    <div
      className="ui-label ui-label--4dx"
    >
      4DX
    </div>
  </div>
  <div
    className="ui-movie-title ui-movie-title--4dx"
  >
    <h4
      className="ui-movie-title-text"
    >
      John Wick 4
    </h4>
  </div>
</div>
`;

exports[`render without label 1`] = `
<div
  className="ui-movie"
>
  <div
    className="ui-movie-img"
  >
    <img
      alt="John Wick 4"
      className="ui-movie-img-asset"
      src="https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/56c72e90-7908-4970-8e1b-b5b7098f4fda/df4gi93-bb27d107-8b1f-4dec-97f1-815424419af6.png/v1/fill/w_1280,h_1811,q_80,strp/john_wick_4___poster_v1_by_pyramid3_df4gi93-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9MTgxMSIsInBhdGgiOiJcL2ZcLzU2YzcyZTkwLTc5MDgtNDk3MC04ZTFiLWI1YjcwOThmNGZkYVwvZGY0Z2k5My1iYjI3ZDEwNy04YjFmLTRkZWMtOTdmMS04MTU0MjQ0MTlhZjYucG5nIiwid2lkdGgiOiI8PTEyODAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.xsATywrHQvyQFggEG8sGHffq7CApdMWZn7FJ9C_CgEA"
    />
  </div>
  <div
    className="ui-movie-title ui-movie-title--undefined"
  >
    <h4
      className="ui-movie-title-text"
    >
      John Wick 4
    </h4>
  </div>
</div>
`;
